FirebaseRemoteConfig/Sources/RCNFetch.m

```diff
@@ -489,23 +494,33 @@ - (void)fetchWithUserProperties:(NSDictionary *)userProperties
     });
   };
 
-  if (gGlobalTestBlock) {
-    gGlobalTestBlock(fetcherCompletion);
+  // If there is a fake config, use that and skip the big block above that does the remote fetch.
+  if ([RCNFakeFetch active]) {
+    self->_settings.isFetchInProgress = NO;
+
+    NSDictionary<NSString *, id> *fakeConfig = [RCNFakeFetch get];
+    // Update config content to cache and DB.
+    if ([fakeConfig[@"state"] isEqualToString:RCNFetchResponseKeyStateUpdate]) {
+      NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
+      [formatter setDateFormat:@"yyyy-MM-dd HH:mm:ss"];
+      self->_settings.lastETag = [formatter stringFromDate:[NSDate date]];
+    }
+    [_content updateConfigContentWithResponse:fakeConfig forNamespace:self->_FIRNamespace];
+    // Update experiments.
+    [_experiment
+        updateExperimentsWithResponse:fakeConfig[RCNFetchResponseKeyExperimentDescriptions]];
+    [self->_settings updateMetadataWithFetchSuccessStatus:YES];
+    completionHandler(FIRRemoteConfigFetchStatusSuccess, nil);
     return;
   }
+
   FIRLogDebug(kFIRLoggerRemoteConfig, @"I-RCN000061", @"Making remote config fetch.");
 
   NSURLSessionDataTask *dataTask = [self URLSessionDataTaskWithContent:compressedContent
                                                      completionHandler:fetcherCompletion];
   [dataTask resume];
 }
 
-+ (void)setGlobalTestBlock:(RCNConfigFetcherTestBlock)block {
```

Comment:
I would double check https://github.com/firebase/firebase-ios-sdk/tree/master/FirebaseRemoteConfig/Tests/Unit
as some of the tests are using this.

Reply:
Confirmed.